### DISCUSSION TITLE
Possibility of adding support for trashy?

### DISCUSSION THREAD

**Discussion by limaceous-bushwhacker** [jarun/nnn#1761] [Q&A] [answered]
Alternative to trash-cli; much better than it actually.

https://github.com/oberblastmeister/trashy

**Comment by jarun** [comment]
We already have support for 2 popular/widely available utilities.
Please modify the code and use it.

**Comment by gurgelgubbe** [comment]
Would be nice if NNN_TRASH was not a set number of options. That it would instead take whatever command the user wants the variable to represent. The manpage could then simply suggest two common utilities and leave it to the user to decide. Future mega-trends of trash management avoided, win.

I could imagine multiple ways a user would want to handle trash besides simply what utility to utilize.

*Hack up or put up* I guess but that makes any user input rather pointless don't you think? ;)

**Reply by limaceous-bushwhacker** [reply]
Seconded, that's a good idea!

**Reply by jarun** [reply]
> That it would instead take whatever command the user wants the variable to represent.

Unfortunately the utilities are not always option-compatible.

**Comment by limaceous-bushwhacker** [comment]
@jarun Any pointers on how to modify the source code to use a different utitlity? Another trash manager has popped up: https://github.com/umlx5h/gtrash

**Reply by jarun** [reply]
Sure! Search for the string "trash" in the code.

**Comment by N-R-K** [comment, marked as answer]
@limaceous-bushwhacker `master` branch now supports custom command in `$NNN_TRASH`. You don't need to patch the source or use some wrapper anymore. See the [wiki](https://github.com/jarun/nnn/wiki/Usage#configuration) and manpage for more info.
